### PR TITLE
dvtm: update to 0.15

### PR DESCRIPTION
**Pull request by herbygillot**
- remove unneeded patches
- use the makefile portgroup
- use the HTTPS URL
- add size checksum field
- use dvtm terminfo files provided by ncurses instead of packaging
dvtm's

###### Tested on

macOS 10.15.7 19H15
Xcode 12.2 12B45b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -d install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/dvtm/Portfile [hunk 1]
```diff
@@ -18,23 +19,31 @@ long_description        dvtm brings the concept of tiling window management, \
                         it easy to work with multiple console based programs \
                         like vim, mutt, cmus or irssi.
 
-homepage                http://brain-dump.org/projects/dvtm/
-master_sites            ${homepage}
+homepage                https://brain-dump.org/projects/dvtm
+master_sites            ${homepage}/
```
Comment (ryandesign) [inline sysutils/dvtm/Portfile:23]: We should use the canonical homepage -- the one the web server redirects to. See https://trac.macports.org/wiki/howto/AvoidRedirects
```suggestion
homepage                https://www.brain-dump.org/projects/dvtm/
master_sites            ${homepage}
```

File: sysutils/dvtm/Portfile [hunk 2]
```diff
@@ -18,23 +19,31 @@ long_description        dvtm brings the concept of tiling window management, \
...
-patchfiles              patch-src-config.mk.diff
+makefile.has_destdir    yes
 
-platform darwin {
-    patchfiles-append   patch-src-Makefile.diff
-}
+build.pre_args-append   PREFIX=${prefix} \
+                        VERSION=${version}
+
+destroot.args           PREFIX=${prefix}
```
Comment (ryandesign) [inline sysutils/dvtm/Portfile:36]: One of the purposes of the makefile 1.0 portgroup is to set PREFIX for you, so you should not need to set it manually. See https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/makefile-1.0.tcl#L113-L125. So you should only need to add the project-specific VERSION, and usually you would do so using `build.args`. `*.pre_args` and `*.post_args` are more likely to have defaults set by MacPorts base and/or portgroups and so it's usually safer not to mess with them in a portfile.

File: sysutils/dvtm/Portfile [hunk 3]
```diff
@@ -18,23 +19,31 @@ long_description        dvtm brings the concept of tiling window management, \
...
-                        CC=${configure.cc}
+platform darwin {
+    configure.cflags-append -D_DARWIN_C_SOURCE
+}
 
-destroot.args           PREFIX=${prefix}
+post-destroot {
+    # dvtm terminfo files are already provided by ncurses,
+    # so do not include them in the destroot
+    file delete -force ${destroot}${prefix}/share/terminfo
```
Comment (ryandesign) [inline sysutils/dvtm/Portfile:48]: Note that `delete` is a more succinct MacPorts synonym for the standard Tcl `file delete -force`.
```suggestion
    delete ${destroot}${prefix}/share/terminfo
```

File: sysutils/dvtm/Portfile [hunk 4]
```diff
@@ -18,23 +19,31 @@ long_description        dvtm brings the concept of tiling window management, \
...
 
-platform darwin {
-    patchfiles-append   patch-src-Makefile.diff
-}
+build.pre_args-append   PREFIX=${prefix} \
+                        VERSION=${version}
+
+destroot.args           PREFIX=${prefix}
 
 use_configure           no
```
Comment (ryandesign) [inline sysutils/dvtm/Portfile:38]: The makefile 1.0 portgroup sets `use_configure no` for you.
```suggestion
```

File: sysutils/dvtm/Portfile (hunk 2 above)
Comment (herbygillot) [reply sysutils/dvtm/Portfile]: So I've run into issues where it seemed as if seemingly the PREFIX for some reason was not being set, and I distinctly remember seeing that when making changes here, but I guess I was wrong.  Changes made.

File: sysutils/dvtm/Portfile (hunk 3 above)
Comment (herbygillot) [reply sysutils/dvtm/Portfile]: Change made

File: sysutils/dvtm/Portfile (hunk 4 above)
Comment (herbygillot) [reply sysutils/dvtm/Portfile]: Good point.  Thanks @ryandesign.

**Pull request merged**